Documentation/capabilities-guide.md

```diff
@@ -319,6 +320,93 @@ image: using image from local store for image name quay.io/coreos/alpine-sh
 listening on [::]:80 ...
 ```
 
+## Capabilities when running as non-root
+
+The capability isolators (and default capabilities) mentioned in this document operate on the capability bounding set.
+When running containers as non-root, capabilities are not added to the effective set of the process, which is the one the kernel checks when doing privileged operations.
+This means the process won't be able to run the privileged operations enabled by the capabilities granted to the container directly.
+
+For example, including `CAP_NET_RAW` in the retain set when running a container as non-root doesn't allow the container to run `ping` (which uses raw sockets):
+
+```
+$ sudo rkt run --interactive kinvolk.io/aci/busybox --user=1000 --group=1000 --caps-retain=CAP_NET_RAW --exec ping -- 8.8.8.8
+PING 8.8.8.8 (8.8.8.8): 56 data bytes
+ping: permission denied (are you root?)
+```
+
+To be able to execute `ping` as a non-root user, the binary needs to have the corresponding file capability.
+
+Also, running an image with file capabilities requires disabling seccomp in rkt.
+This is due to a systemd bug where using seccomp results in enabling [no_new_privs][NNP] (you can track progress in [systemd/systemd#6763](https://github.com/systemd/systemd/pull/6763)).
```

Comment:
There should be an issue in rkt for this, matching the one in systemd, because when it gets fixed in systemd, it will not be automatically fixed in rkt...

-Also, running an image with file capabilities requires disabling seccomp in rkt.
+**Note:** running an image with file capabilities **currently** requires disabling seccomp in rkt.